controllers/transforms_test.go

```diff
@@ -444,8 +445,8 @@ func TestTransformForRuntime(t *testing.T) {
 					},
 				}),
 			expectedOutput: NewDaemonset().
-				WithHostPathVolume("containerd-config", filepath.Dir(DefaultContainerdConfigFile), newHostPathType(corev1.HostPathDirectoryOrCreate)).
-				WithHostPathVolume("containerd-drop-in-config", "/etc/containerd/conf.d", newHostPathType(corev1.HostPathDirectoryOrCreate)).
+				WithHostPathVolume("containerd-config", filepath.Dir(DefaultContainerdConfigFile), internalptr.HostPathType(corev1.HostPathDirectoryOrCreate)).
```

Comment:
Question -- is there a technical reason for not using the methods provided in `k8s.io/utils/ptr`? For example, could we not replace this with:
```suggestion
				WithHostPathVolume("containerd-config", filepath.Dir(DefaultContainerdConfigFile), ptr.To(corev1.HostPathDirectoryOrCreate)).
```
which would eliminate the need for our own internal `ptr` package?

Reply:
Great catch, I think you're right. I don't think we need our own `ptr` package. 

I'm not sure why we had these original pointer methods. I guess that means we don't need `newHostPathType` at all.  I've updated the code.

Reply:
Sweet, thanks!

Can you update the PR title? After getting another pair of eyes on this, we can squash then merge.